users/test/test_views.py

```diff
@@ -7,12 +7,108 @@
 from rest_framework.test import APIRequestFactory, APITestCase
 
-from users.views import PasswordResetAPIView, PasswordResetEmailView
+from users.views import (
+    PasswordResetAPIView,
+    PasswordResetEmailView,
+    UserDetail,
+    UserList,
+)
 
 fake = Faker()
 User = get_user_model()
+
 password_reset_email_view = PasswordResetEmailView.as_view()
 password_reset_api_view = PasswordResetAPIView.as_view()
+user_detail = UserDetail.as_view()
+user_list = UserList.as_view()
+fake = Faker()
+
+
+class TestUserList(APITestCase):
+    def setUp(self) -> None:
+        self.user = User.objects.create_user(
+            username=fake.user_name(),
+            email=fake.email(),
+            password=fake.password(),
+        )
+        self.factory = APIRequestFactory()
+
+    def test_get_user_list(self) -> None:
+        url = reverse("users")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_create_user(self) -> None:
+        url = reverse("users")
+        data = {
+            "username": fake.user_name(),
+            "email": fake.email(),
+            "password": fake.password(),
+        }
+        response = self.client.post(url, data=data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(User.objects.count(), 2)
+
+    def test_create_user_with_existing_email(self) -> None:
+        url = reverse("users")
+        data = {
+            "username": fake.user_name(),
+            "email": self.user.email,
+            "password": fake.password(),
+        }
+        response = self.client.post(url, data=data)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+
+class TestUserDetail(APITestCase):
+    def setUp(self) -> None:
+        self.user = User.objects.create_user(
+            username=fake.user_name(),
+            email=fake.email(),
+            password=fake.password(),
+        )
+        self.factory = APIRequestFactory()
+
+    def test_get_user_detail(self) -> None:
+        url = reverse("user-detail", kwargs={"lookup_id": self.user.lookup_id})
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_user_detail_with_invalid_id(self) -> None:
+        url = reverse("user-detail", kwargs={"lookup_id": -1})
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_user(self) -> None:
+        url = reverse("user-detail", kwargs={"lookup_id": self.user.lookup_id})
+        data = {
+            "username": fake.user_name(),
+            "email": fake.email(),
+            "password": fake.password(),
+        }
+        response = self.client.put(url, data=data)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_update_user_with_invalid_id(self) -> None:
```

Comment:
Add test for when a user tries to update another persons account

Reply:
Also add that a superuser is allowed to edit/delete make sure your tests are comprehensive as possible